src/entities/communitySbt.ts

```diff
@@ -553,80 +539,105 @@ class SBT {
    * ranking of all users. Check if given user is in top 5.
    * If so, assign a top trader/LP badge, otherwise return undefined
    */
-  public async getTopBadge(
+  public async getTopBadges(
     userId: string,
     seasonId: number,
-    isLP: boolean,
     seasonStart: number,
     seasonEnd: number,
     selectedBadgesSubgraphUrl?: string,
-  ): Promise<BadgeResponse | undefined> {
-    const badgeType = getTopBadgeType(seasonId, !isLP);
-    if (!badgeType) return undefined;
+  ): Promise<{
+    topTraderBadge: BadgeResponse | undefined;
+    topLpBadge: BadgeResponse | undefined;
+  }> {
+    const traderBadgeType = getTopBadgeType(seasonId, true);
+    const lpBadgeType = getTopBadgeType(seasonId, false);
 
     if (!selectedBadgesSubgraphUrl || !this.coingeckoKey || !this.ignoredWalletIds) {
-      return undefined;
+      return {
+        topTraderBadge: undefined,
+        topLpBadge: undefined,
+      };
     }
 
     if (!this.ethPrice) {
       this.ethPrice = await geckoEthToUsd(this.coingeckoKey);
     }
 
-    const rankResult = await this.getRanking({
-      seasonStart,
-      seasonEnd,
-      isLP,
-    });
+    const rankResult = await this.getRanking(seasonId);
 
-    for (let rank = 0; rank < 5; rank++) {
-      if (!rankResult[rank]) {
-        return undefined;
-      }
-      const entry = rankResult[rank];
-      if (entry.address === userId) {
-        const badge = await this.constructTopBadge(
+    const traderPosition = rankResult.traderRankResults
+      .slice(0, 5)
+      .find((rank) => rank.address.toLowerCase() === userId.toLowerCase());
+
+    const topTraderBadge = !isUndefined(traderPosition)
+      ? await SBT.constructTopBadge(
```

Comment:
why SBT.f() instead of this.f()?

Reply:
because the function is ``static`` at this point